.github/workflows/test.yml

```diff
@@ -2,59 +2,37 @@ name: 'Test'
 
 on: [push, pull_request, workflow_dispatch]
 
+
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ${{ matrix.os }}
     timeout-minutes: 10
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        python-version: ['3.8', '3.9', '3.10']
 
     steps:
-      - name: Check out code
-        uses: actions/checkout@v2
-
-      - name: Setup Python
-        uses: actions/setup-python@v2
-        with:
-          python-version: "3.9"
-
-      - name: Cache pip
-        uses: actions/cache@v2
-        with:
-          path: ~/.cache/pip
-          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
-          restore-keys: |
-            ${{ runner.os }}-pip-
-
-      - name: Install dependencies
-        run:
-          python -m pip install -r tests/requirements.txt
-
-      - name: Run pyright tests
-        uses: gramster/pyright-action@main
-        with:
-          project: pyrighttestconfig.json
-          warn-partial: true
-
-      - name: Run pytest
-        run:
-          pytest tests/pandas
-          
-      - name: Run mypy
-        run:
-          mypy tests/pandas typings/pandas
-          
-      - name: Build wheel and install and remove typings
-        run: |
-          python setup.py build bdist_wheel
-          find ./dist/*.whl | xargs pip install
-          rm -rf typings
-          pip install pyright
-      
-      - name: Run pyright against dist
-        uses: gramster/pyright-action@main
-        with:
-          project: pyrightdistconfig.json
-          warn-partial: true
-
-      - name: Run mypy against dist
-        run:
-          mypy tests/pandas
```

Comment:
Can you add these steps (lines 45-60) back, so that we see how to build the wheel with `poetry`, install the wheel, and run `pyright` and `mypy` against the installed version.